float32 mean_x
float32 mean_y
float32 cov_x
float32 cov_y
